src/RazorSdk/SourceGenerators/RazorSourceGenerator.cs

```diff
@@ -278,11 +278,16 @@ private static string GetIdentifierFromPath(string filePath)
 
             for (var i = 0; i < filePath.Length; i++)
             {
-                builder.Append(filePath[i] switch
+                switch (filePath[i])
                 {
-                    ':' or '\\' or '/' => '_',
-                    var @default => @default,
-                });
+                    case ':' or '\\' or '/':
```

Comment:
Is this still required when we have the below `!char.IsLetterOrDigit`?

Reply:
Not really -- I mostly kept it so that it was clear treating path separators differently.